extensions/sql-migration/src/api/dataModels/azure/networkInterfaceModel.ts

```diff
@@ -0,0 +1,147 @@
+	id: string,
+	name: string,
+	type: string,
+	location: string,
+	properties: any
+}
+
+export interface NetworkInterface extends NetworkResource {
+	properties: {
+		ipConfigurations: NetworkInterfaceIpConfiguration[],
+		primary: boolean,
+		provisioningState: string,
+		resourceGuid: string,
+		virtualMachine: {
+			id: string,
+		},
+	},
+}
+
+export interface NetworkInterfaceIpConfiguration extends NetworkResource {
+	properties: {
+		primary: boolean,
+		privateIPAddress: string,
+		privateIPAddressVersion: string,
+		provisioningState: string,
+		publicIPAddress: NetworkResource
+	}
+}
+
+export interface PublicIpAddress extends NetworkResource {
+	properties: {
+		ipAddress: string
+	}
+}
+
+export class NetworkInterfaceModel {
+	public static IPv4VersionType = "IPv4".toLocaleLowerCase();
+
+	public static getPrimaryNetworkInterface(networkInterfaces: NetworkInterface[]): NetworkInterface | undefined {
+		if (networkInterfaces && networkInterfaces.length > 0) {
+			const primaryNetworkInterface = networkInterfaces.find(nic => nic.properties.primary === true);
+			if (primaryNetworkInterface) {
+				return primaryNetworkInterface;
+			}
+		}
+
+		return undefined;
+	}
+
+	public static getPrimaryIpConfiguration(networkInterface: NetworkInterface): NetworkInterfaceIpConfiguration | undefined {
+		const hasIpConfigurations = networkInterface?.properties?.ipConfigurations && networkInterface.properties.ipConfigurations?.length > 0;
+		if (!hasIpConfigurations) {
+			return undefined;
+		}
+
+		// If the primary property exists, return the primary, otherwise, return the first ip configuration
+		let primaryIpConfig = networkInterface.properties.ipConfigurations.find((ipConfig: NetworkInterfaceIpConfiguration) => ipConfig.properties.primary);
+		if (primaryIpConfig) {
+			return primaryIpConfig;
+		}
+
+		// Otherwise, find the first configuration with a public ip address.
+		primaryIpConfig = networkInterface.properties.ipConfigurations.find(ipConfiguration => ipConfiguration.properties.publicIPAddress !== undefined);
+		if (primaryIpConfig) {
+			return primaryIpConfig;
+		}
+
+		// Otherwise, return the first ipv4 configuration
+		primaryIpConfig = networkInterface.properties.ipConfigurations.find(ipConfiguration => ipConfiguration.properties.privateIPAddressVersion.toLocaleLowerCase() === NetworkInterfaceModel.IPv4VersionType);
+		return primaryIpConfig;
+	}
+
+	public static getIpAddress(networkInterfaces: NetworkInterface[]): string {
+		const primaryNetworkInterface = this.getPrimaryNetworkInterface(networkInterfaces);
+
+		if (!primaryNetworkInterface) {
+			return "";
+		}
+
+		const ipConfig = this.getPrimaryIpConfiguration(primaryNetworkInterface);
+		if (ipConfig && ipConfig.properties.publicIPAddress) {
+			return ipConfig.properties.publicIPAddress.properties.ipAddress;
+		}
+
+		if (ipConfig && ipConfig.properties.privateIPAddress) {
+			return ipConfig.properties.privateIPAddress;
+		}
+
+		return "";
```

Comment:
if return "", it is an error?

Reply:
yeah, so it returns empty string. The handlers will error and report the error. 

Do you have any suggestions on how to handle this better? I'm open to ideas

Reply:
@AkshayMata just curious what scenarios can lead to line 98? may be add some logging here?